### ISSUE TITLE
Long.compareUnsigned does not work

### ISSUE THREAD

**Issue by warownia1** [BobHanson/java2script#270]
The implementation of `Long.compareUnsigned` utilizes long overflow for its proper function which does not work as intended in java2script. For all positive values I tested for, the method always returns 0. For negative values, it returns the number opposite to what would be expected.

Related to https://github.com/BobHanson/java2script/issues/202

**Comment by BobHanson** [comment]
This is fixed but not pushed. The issue was also found in an trial to use BigInteger and BigDecimal for an algorithm that calculates the digits of pi. We found that after about 980 digits, anomalies arose. This was tracked down to the presence of two Long.signum$J() methods present in j2sClazz.js, with the second (original, simple) one overriding the first (new, correct) one. This now should work correctly in all cases. test.Test_Long.java has been updated to test for correct comparison, and JavaScript assertion succeeds. The algorithm correctly generates pi to at least the precision of 10000 digits. It was not checked for more than that. 

Efforts are now underway to speed up BigInteger and BigDecimal in JavaScript by removing Longs altogether from the calculations, but this is not a high priority. See issue #271
